wallet-client/Main.hs

```diff
@@ -54,26 +58,26 @@ instance Exception PartialConfigError
 main :: IO ()
 main = do
 
-    cardanoConfiguration <-
-      case finaliseCardanoConfiguration mainnetConfiguration of
-        Right cc -> return cc
-        Left err -> throwIO (PartialConfigError err)
     cardanoEnvironment  <- initializeCardanoEnvironment
 
     logConfig           <- execParser opts
 
-    (cardanoFeatures, nodeLayer) <- initializeAllFeatures logConfig cardanoConfiguration cardanoEnvironment
+    (cardanoFeatures, nodeLayer) <- initializeAllFeatures logConfig mainnetConfiguration cardanoEnvironment
 
     let cardanoApplication :: NodeLayer -> CardanoApplication
         cardanoApplication = CardanoApplication . nlRunNode
 
     runCardanoApplicationWithFeatures Development cardanoFeatures (cardanoApplication nodeLayer)
 
-initializeAllFeatures :: ArgParser -> CardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
-initializeAllFeatures (ArgParser logCli cli) cardanoConfiguration cardanoEnvironment = do
+initializeAllFeatures :: ArgParser -> PartialCardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
+initializeAllFeatures (ArgParser logCli cli) partialConfig cardanoEnvironment = do
+
+    finalConfig <- either (throwIO . ConfigurationError) pure $
+                   finaliseCardanoConfiguration $
+                   mergeConfiguration partialConfig (cliGenesis cli) (cliKeyMaterial cli)
```

Comment:
I think it's better to put this way:
```
case finaliseCardanoConfiguration ... of
  Left err -> throwIO (ConfigurationError err)
  Right finalConfig -> ...
```

Reply:
Done, as well as in `cardano-node`'s case.